src/ngraph/pass/constant_folding_scatter.cpp

```diff
@@ -0,0 +1,301 @@
+
+        runtime::reference::scatter_elem_update<DataType, IndicesType>(
+            data->get_data_ptr<DataType>(),
+            indices->get_data_ptr<IndicesType>(),
+            updates->get_data_ptr<DataType>(),
+            normalized_axis,
+            data_ptr,
+            data->get_shape(),
+            indices->get_shape());
+    }
+    else
+    {
+        throw ngraph_error("Unsupported op in scatter_elem_updt constant folding.");
+    }
+
+    return make_shared<op::Constant>(
+        scatter->get_output_element_type(0), scatter->get_output_shape(0), data_ptr);
+}
+
+template <typename T, typename U>
+static shared_ptr<op::Constant>
+    dispatch_const_fold_indices(const shared_ptr<op::Constant>& data,
+                                const shared_ptr<op::Constant>& indices,
+                                const shared_ptr<op::Constant>& updates,
+                                const shared_ptr<op::Constant>& axis,
+                                const shared_ptr<Node>& scatter_elem_updt)
+{
+    auto axis_type = axis->get_output_element_type(0);
+
+    // Dispatch specialization based on axis data type.
+    switch (axis_type)
+    {
+    case element::Type_t::undefined:
+        NGRAPH_CHECK(false,
+                     "Encountered 'undefined' element type in constant_scatter_elem_updt_callback");
+        break;
+    case element::Type_t::dynamic:
+        NGRAPH_CHECK(false,
+                     "Encountered 'dynamic' element type in constant_scatter_elem_updt_callback");
+        break;
+    case element::Type_t::u8:
+        return fold_constant_scatter_elem_updt<T, U, uint8_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u16:
+        return fold_constant_scatter_elem_updt<T, U, uint16_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u32:
+        return fold_constant_scatter_elem_updt<T, U, uint32_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u64:
+        return fold_constant_scatter_elem_updt<T, U, uint64_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i8:
+        return fold_constant_scatter_elem_updt<T, U, int8_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i16:
+        return fold_constant_scatter_elem_updt<T, U, int16_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i32:
+        return fold_constant_scatter_elem_updt<T, U, int32_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i64:
+        return fold_constant_scatter_elem_updt<T, U, int64_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::boolean:
+    case element::Type_t::bf16:
+    case element::Type_t::f16:
+    case element::Type_t::f32:
+    case element::Type_t::f64:
```

Comment:
Just a thought, but scatter does no math on the values so it should work by just scattering the byte-size of the value. You really don't need to look at the full type. No need to change anything if you don't want.

Reply:
Good point. Tried to implement this idea at dispatch points. Pleas take a look if this is what you mean.